apps/op-func/src/infra/storage/access-log-register.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  BlobServiceClient,
+  BlockBlobUploadResponse,
+  ContainerClient,
+} from "@azure/storage-blob";
+import {
+  AuditEvent,
+  auditEventSchema,
+} from "io-fims-common/domain/audit-event";
+import * as assert from "node:assert/strict";
+
+export class AccessLogRegister {
+  #container: ContainerClient;
+
+  constructor(blobServiceClient: BlobServiceClient, containerName: string) {
+    this.#container = blobServiceClient.getContainerClient(containerName);
+  }
+
+  // A helper method used to read a Node.js readable stream into a String
+  async #streamToAuditEvent(readableStream: NodeJS.ReadableStream) {
+    let result = "";
+    for await (const chunk of readableStream) {
+      result += chunk;
+    }
+
+    try {
+      const parsedResult = JSON.parse(result);
+      return auditEventSchema.parse(parsedResult);
+    } catch (e) {
+      throw new Error(`Error during conversion of the downloaded stream`, {
+        cause: e,
+      });
+    }
+  }
```

Comment:
this functions breaks SRP, because it does two things:
1. "unwraps" a `ReadableStream` to a string
2. parse the resulting string as `auditEventSchema`

imho the parsing part should be moved into the `get` method